src/umzug.ts

```diff
@@ -61,17 +62,11 @@ export class Umzug extends EventEmitter {
 
 	// #region EventEmitter explicit implementation typings
 
-	on(eventName: UmzugEventNames, cb?: (name: string, migration: Migration) => void): this {
-		return super.on(eventName, cb);
-	}
+	on: (eventName: UmzugEventNames, cb?: (name: string, migration: Migration) => void) => this;
 
-	addListener(eventName: UmzugEventNames, cb?: (name: string, migration: Migration) => void): this {
-		return super.addListener(eventName, cb);
-	}
+	addListener: (eventName: UmzugEventNames, cb?: (name: string, migration: Migration) => void) => this;
 
-	removeListener(eventName: UmzugEventNames, cb?: (name: string, migration: Migration) => void): this {
-		return super.removeListener(eventName, cb);
-	}
+	removeListener: (eventName: UmzugEventNames, cb?: (name: string, migration: Migration) => void) => this;
```

Comment:
Since this extending EventEmitter already, all we really need to do is override the types rather than the implementation. But I might undo this change and put it in a later PR since it's not really related to the rest.

Reply:
reverted